balancer-labs/frontend-v2
# Add Gnosis Safe connector

# Description

This PR adds a new Gnosis Safe connector which allows loading the interface through the Gnosis website to directly connect to a Safe.

## Type of change


## How should this be tested?

The main issue which needs to be tested is that we correctly pick up when a transaction confirms when using a Gnosis safe with only one signer. This is because we don't talk directly with the RPC but through a Gnosis service which sometimes can not know about a transaction that's in flight for some time.

Testing that you can do a single transaction and not have the app get stuck in an infinite load should be sufficient but we should probably test each transaction type just to be thorough.

You can load the app on either [mainnet](https://gnosis-safe.io) or [polygon](https://polygon.gnosis-safe.io)

To open as a safe app:
- Load a Safe on https://polygon.gnosis-safe.io and go to the Apps tab.
- Click the first tile on that page which allows loading a custom safe app.
- Set the most recent preview URL as the URL and the name+icon should autofill. Click add.
- You'll be taken to the app running in an iframe (It's possible that your wallet will autoload at this point, just log out if so)
- Connect again using the new option with the name "Gnosis Safe".

**If you're trying to run this on mainnet then you need to run the app locally as passwords on Vercel prevent it from loading properly**

Known issues:
- ~~Transactions in the "recent activity" menu use the "safeTxHash" rather than the real hash so the links are broken.~~ Addressed
- ~~It's possible to connect to metamask on the Gnosis interface so we should detect this case and prevent connecting to other wallets.~~ Addressed

## Checklist:
